ZAPD/Globals.cpp

```diff
@@ -82,47 +98,67 @@ void Globals::ReadConfigFile(const std::string& configFilePath)
 	if (root == nullptr)
 		return;
 
-	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != NULL;
+	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != nullptr;
 	     child = child->NextSiblingElement())
 	{
-		if (std::string(child->Name()) == "SymbolMap")
-		{
-			std::string fileName = std::string(child->Attribute("File"));
-			GenSymbolMap(Path::GetDirectoryName(configFilePath) + "/" + fileName);
-		}
-		else if (std::string(child->Name()) == "Segment")
+		auto it = ConfigTypeDictionary.find(child->Name());
+		if (it != ConfigTypeDictionary.end())
 		{
-			std::string fileName = std::string(child->Attribute("File"));
-			int32_t segNumber = child->IntAttribute("Number");
-			segmentRefs[segNumber] = fileName;
-		}
-		else if (std::string(child->Name()) == "ActorList")
-		{
-			std::string fileName = std::string(child->Attribute("File"));
-			std::vector<std::string> lines =
-				File::ReadAllLines(Path::GetDirectoryName(configFilePath) + "/" + fileName);
+			switch (it->second)
+			{
+				case ConfigType::SYMBOL_MAP:
+				{
+					std::string fileName = child->Attribute("File");
+					GenSymbolMap(Path::GetDirectoryName(configFilePath) + "/" + fileName);
+				}
+				break;
 
-			for (std::string line : lines)
-				cfg.actorList.push_back(StringHelper::Strip(line, "\r"));
-		}
-		else if (std::string(child->Name()) == "ObjectList")
-		{
-			std::string fileName = std::string(child->Attribute("File"));
-			std::vector<std::string> lines =
-				File::ReadAllLines(Path::GetDirectoryName(configFilePath) + "/" + fileName);
+				case ConfigType::SEGMENT:
+				{
+					std::string fileName = child->Attribute("File");
+					int32_t segNumber = child->IntAttribute("Number");
+					segmentRefs[segNumber] = fileName;
+				}
+				break;
 
-			for (std::string line : lines)
-				cfg.objectList.push_back(StringHelper::Strip(line, "\r"));
-		}
-		else if (std::string(child->Name()) == "TexturePool")
-		{
-			std::string fileName = std::string(child->Attribute("File"));
-			ReadTexturePool(Path::GetDirectoryName(configFilePath) + "/" + fileName);
+				case ConfigType::ACTOR_LIST:
+				{
+					std::string fileName = child->Attribute("File");
+					std::vector<std::string> lines =
+						File::ReadAllLines(Path::GetDirectoryName(configFilePath) + "/" + fileName);
+
+					for (std::string line : lines)
+						cfg.actorList.push_back(line);
```

Comment:
Unlikely to matter in practice, but note that this performs a double copy (`line` is copied from the vector, and then a second copy occurs when it is inserted into cfg.actorList). You can eliminate both copies (and allocations) by doing:

```cpp
for (auto& line : lines)
  cfg.actorList.emplace_back(std::move(line));
```